File: Core/src/main/java/dev/lynith/core/utils/Key.java
package dev.lynith.core.utils;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.Objects;

@RequiredArgsConstructor
public class Key {

    @Getter
    private final int code;

    @Override
    public String toString() {
        return "Key{" +
                "code=" + code +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Key)) return false;
        Key key = (Key) o;
        return code == key.code;
    }

    @Override
    public int hashCode() {
        return Objects.hash(code);
    }
}

File: Core/src/main/kotlin/dev/lynith/core/input/Key.kt
package dev.lynith.core.input

class Key(
    val keyCode: Int
) {

    override fun toString(): String {
        return "Key(keyCode=$keyCode)"
    }

    override fun equals(other: Any?): Boolean {
        if (other !is Key) return false
        if (this === other) return true
        return keyCode == other.keyCode
    }

}